auto_nag/scripts/crash_samll_volume.py

```diff
@@ -0,0 +1,188 @@
+        new_criteria = []
+        for criterion in topcrash_criteria:
+            criterion = {
+                **criterion,
+                "tc_limit": criterion["tc_limit"] * factor,
+            }
+            if "tc_startup_limit" in criterion:
+                criterion["tc_startup_limit"] = criterion["tc_startup_limit"] * factor
+
+            new_criteria.append(criterion)
+
+        return new_criteria
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        existing_keywords = {
+            keyword
+            for keyword in ("topcrash", "topcrash-startup")
+            if keyword in bug["keywords"]
+        }
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+
+        top_crash_signatures = [
+            signature
+            for signature in signatures
+            if signature in self.topcrash_signatures
+        ]
+
+        keep_topcrash_startup = any(
+            any(
+                criterion["is_startup"]
+                for criterion in self.topcrash_signatures[signature]
+            )
+            for signature in top_crash_signatures
+        )
+
+        keywords_to_remove = None
+        if not top_crash_signatures:
+            keywords_to_remove = existing_keywords - {"topcrash", "topcrash-startup"}
+        elif not keep_topcrash_startup:
+            keywords_to_remove = existing_keywords - {"topcrash-startup"}
+
+        data[bugid] = {
+            "severity": bug["severity"],
+            "keep_topcrash": bool(top_crash_signatures),
+            "keywords_to_remove": keywords_to_remove,
+            "signatures": signatures,
+        }
+
+        return bug
+
+    def _get_low_volume_crash_signatures(self, bugs):
+        signatures = {
+            signature
+            for bug in bugs.values()
+            if not bug["keep_topcrash"] and bug["severity"] in HIGH_SEVERITY
+            for signature in bug["signatures"]
+        }
+
+        signature_volume = Topcrash().fetch_signature_volume(signatures)
+
+        low_volume_signatures = {
+            signature for signature, volume in signature_volume.items() if volume < 5
+        }
+
+        return low_volume_signatures
+
+    def get_bugs(self, date="today", bug_ids=..., chunk_size=None):
+        bugs = super().get_bugs(date, bug_ids, chunk_size)
+        self.set_autofix(bugs)
+
+        # Keep only bugs with an autofix
+        bugs = {
+            bugid: bug for bugid, bug in bugs.items() if bugid in self.autofix_changes
+        }
+
+        return bugs
+
+    def set_autofix(self, bugs):
+        """Set the autofix for each bug."""
+
+        low_volume_signatures = self._get_low_volume_crash_signatures(bugs)
+
+        for bugid, bug in bugs.items():
+            autofix = {}
+            reasons = []
+            if bug["keywords_to_remove"]:
+                reasons.append(
+                    "The crash signature linked to this bug is not a topcrash signature anymore."
```

Comment:
Maybe we could add a link to the rules

Reply:
Done in d36aa0f257c9542bece01df5ab6c5406c80aa19b